mantidproject/mantid
# Updating muon alalysis to work with history

**Description of work.**

This updates the Muon Analysis workflow to have a valid workspace history. This is mainly done by doing all operations via algorithms rather than through python scripting. The final step of creating the TableWorkspace is not currently workspace history compatible as there is no way to create a filled generic table workspace via algorithms currently. The solution to this is to create a new Algorithm CreateTableWorkspace which does this but this has been left for another PR. 



**To test:**
* Open Muon Analysis 2, load some data and perform a fit on it. 
* Wait for a recovery file to be created (these are generally done every minute but the frequency can be changed in the settings)
* Segfault mantid using the Segfault algorithm
* Re-open mantid and check that project recovery proceeds correctly returning the ADS to the state it was in before the crash.
* Check that when a results table is created everything except this is still recovered.



*There is no associated issue.*



---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
